autogen.pl

```diff
@@ -776,6 +776,40 @@ sub find_and_delete {
 # Find a specific executable and ensure that it is a recent enough
 # version.
 
+sub check_m4_gnu_extensions {
+    verbose "   Checking whether m4 supports the GNU extensions\n";
+    my $m4 = $ENV{'M4'};
+    $m4 = "m4" unless (defined $m4);
+    my $cmd = "$m4 -g config/check_m4_gnu_extension.m4";
+    my $result = `$cmd`;
+    if ($? != 0) {
+        $cmd = "$m4 config/check_m4_gnu_extension.m4";
+        $result = `$cmd`;
+    }
+    chomp($result);
+    if ($result eq "ten") {
+        verbose "     ==> SUPPORTED\n";
+    } else {
+        verbose "     ==> NOT SUPPORTED\n";
+        print "\nI am gonna abort.  :-(
+$m4 must support GNU extensions
+
+$cmd
+returned and expected result
+\"$result\", expected value is \"ten\", and known unexpected value is \"one0\"
+
+if you are running on OpenBSD, you might want to
+export M4=gm4
+and run autogen.pl again
+=================================================================\n";
```

Comment:
I suggest this:

```
I need an m4 that supports GNU extensions; sorry!  I am gonna abort.

    $cmd

returned \"$result\", but I need it to return \"ten\".  If you are 
running on OpenBSD (e.g., if the result is \"one0\"), you might want to

export M4=gm4

and try running autogen.pl again.
```

(word wrap that above paragraph appropriately; it's hard to tell ~72 chars here in the github web UI)